components/Primitives/src/StaggeredLayout/StaggeredLayout.cs

```diff
@@ -163,7 +163,7 @@ protected override Size MeasureOverride(VirtualizingLayoutContext context, Size
         if (ItemsStretch is StaggeredLayoutItemsStretch.None)
         {
             columnWidth = double.IsNaN(DesiredColumnWidth) ? availableWidth : Math.Min(DesiredColumnWidth, availableWidth);
-            numColumns = Math.Max(1, (int)Math.Floor(availableWidth / state.ColumnWidth));
+            numColumns = Math.Max(1, (int)Math.Floor(availableWidth / (DesiredColumnWidth + ColumnSpacing)));
```

Comment:
Oh @Poker-sang just noticed, originally this was just `columnWidth` from above, should we still be using that as it does the NaN check? 

We'd just move the `ColumnSpacing` as part of that value on 165? I guess that was missing before? (Maybe it'd be better to move that add of the ColumnSpacing change to #573, and just keep this change to be the fix for the change from `state.`?) Then it's easy to approve the variable substitution here quick for the crash, and we can just re-validate the other behavior changes all within #573 together?